src/mesh/X3D_Draco_Mesh_Reader.hh

```diff
@@ -0,0 +1,133 @@
+//----------------------------------*-C++-*----------------------------------//
+/*!
+ * \file   mesh/X3D_Draco_Mesh_Reader.cc
+ * \author Ryan Wollaeger <wollaeger@lanl.gov>, Kendra Keady
+ * \date   Wednesday, Jul 11, 2018, 14:24 pm
+ * \brief  X3D_Draco_Mesh_Reader header file.
+ * \note   Copyright (C) 2018 Los Alamos National Security, LLC.
+ *         All rights reserved. */
+//---------------------------------------------------------------------------//
+
+#ifndef rtt_mesh_X3D_Draco_Mesh_Reader_hh
+#define rtt_mesh_X3D_Draco_Mesh_Reader_hh
+
+#include "Draco_Mesh_Reader.hh"
+#include "ds++/Assert.hh"
+#include "ds++/config.h"
+#include <map>
+#include <string>
+#include <vector>
+
+namespace rtt_mesh {
+
+//===========================================================================//
+/*!
+ * \class X3D_Draco_Mesh_Reader
+ *
+ * \brief Read x3d file format and supply data to Draco_Mesh_Builder.
+ *
+ * This class parses mesh data from an x3d file format and manipulates it into a
+ * form that is compatible to the Draco_Mesh_Builder class.  The parsing of the
+ * file stream to the map follows work by Kendra Keady on a parsing framework.
+ *
+ * \todo: Boundary condition data is evidently parsed separately in X3D, so
+ * meshes generated from this reader will not have side flag data (which ids
+ * boundary conditions.
+ *
+ * \todo: Consider using the Class_Parse_Table formalism developed by Kent Budge
+ * as an alternative.
+ */
```

Comment:
Provide a link to X3D specification?

Reply:
Should this be to the pdf on the confluence page?

Reply:
Yes - the link should point to the confluence page.  I don't think the document has an LA-UR.
It is probably worth the effort to add a copy at github.com/Draco/dracodoc.git.